internal/convert.go

```diff
@@ -256,12 +258,36 @@ func (conv *Conv) SampleBadRows(n int) []string {
 func (conv *Conv) AddPrimaryKeys() {
 	for t, ct := range conv.SpSchema {
 		if len(ct.Pks) == 0 {
-			k := conv.buildPrimaryKey(t)
-			ct.ColNames = append(ct.ColNames, k)
-			ct.ColDefs[k] = ddl.ColumnDef{Name: k, T: ddl.Type{Name: ddl.Int64}}
-			ct.Pks = []ddl.IndexKey{{Col: k}}
+			primaryKeyPopulated := false
+			// Populating column with unique constraint as primary key in case
+			// table doesn't have primary key and removing the unique index.
+			if len(ct.Indexes) != 0 {
+				for i, index := range ct.Indexes {
+					if index.Unique {
+						for _, indexKey := range index.Keys {
+							ct.Pks = append(ct.Pks, ddl.IndexKey{Col: indexKey.Col, Desc: indexKey.Desc})
+							conv.UniquePKey[t] = append(conv.UniquePKey[t], indexKey.Col)
+						}
+						primaryKeyPopulated = true
+						if len(ct.Indexes) == 1 {
+							ct.Indexes = nil
+						} else {
+							tempIndex := ct.Indexes[:i]
+							tempIndex = append(tempIndex, ct.Indexes[i+1:]...)
+							ct.Indexes = tempIndex
```

Comment:
nit: This piece works correctly but I think we can refactor this whole if-else block with `ct.Indexes = append(ct.Indexes[:i], ct.Indexes[i+1:]...)` for better readability. Even when ct.Indexes is singleton, this would yield an empty list which is consistent with other places.

Reply:
This block is nested to 8 levels. Makes it pretty difficult to read.
Suggest using the approach suggested here, or moving the block out into a new method

Reply:
Modified the code block to use suggested approach.